### ISSUE TITLE
NVDA no longer reads the elements in the ribbons in Word 2013

### ISSUE THREAD

**Issue by netblue44** [nvaccess/nvda#5504]
Hi,
Since the versions of NVDA after 2015.3, NVDA does not read the elements in the ribbons in word 2013. For example it reads the names of the ribbons but when they are open it does not read the options, also the ribbons are not red when using the shortcut keys. However it reads the ribbons fine in the other office 2013 applications.
I do not have this problem when using JAWS nor Window eyes.
I am using Windows 7 Ultimate 64 bits.

**Comment by netblue44** [comment]
This is the log file that I was able to get for this problem.

INFO - __main__ (12:40:16):
Starting NVDA
INFO - core.main (12:40:17):
Config dir: C:\Users\USUARIO\AppData\Roaming\nvda
INFO - core.main (12:40:17):
NVDA version next-12845,937f433
INFO - core.main (12:40:17):
Using Windows version 6.1.7601 service pack 1 workstation
INFO - core.main (12:40:17):
Using Python version 2.7.10 (default, May 23 2015, 09:40:32) [MSC v.1500 32 bit (Intel)]
INFO - core.main (12:40:17):
Using comtypes version 0.6.2
INFO - external:synthDrivers.eloquence.SynthDriver.__init__ (12:40:17):
Using Eloquence version 6.1.0.0
INFO - synthDriverHandler.setSynth (12:40:17):
Loaded synthDriver eloquence
INFO - core.main (12:40:17):
Using wx version 3.0.2.0 msw (classic)
INFO - braille.initialize (12:40:17):
Using liblouis version 2.6.4
INFO - braille.BrailleHandler.setDisplayByName (12:40:17):
Loaded braille display driver noBraille, current display has 0 cells.
INFO - brailleInput.initialize (12:40:17):
Braille input initialized
WARNING - core.main (12:40:17):
Java Access Bridge not available
DEBUGWARNING - inputCore.InputManager.loadLocaleGestureMap (12:40:17):
No locale gesture map for language en
DEBUGWARNING - inputCore.InputManager.loadUserGestureMap (12:40:17):
No user gesture map
DEBUGWARNING - touchHandler.initialize (12:40:17):
Touch only supported on Windows 8 and higher
INFO - core.main (12:40:17):
NVDA initialized
IO - speech.speak (12:40:17):
Speaking [u'Taskbar']
IO - speech.speak (12:40:18):
Speaking [u'Men\xfa Inicio']
IO - speech.speak (12:40:19):
Speaking [u'Cuadro de b\xfasqueda  edit   Buscar programas y archivos']
IO - speech.speak (12:40:19):
Speaking [u'blank']
IO - inputCore.InputManager.executeGesture (12:40:19):
Input: kb(laptop):downArrow
IO - speech.speak (12:40:19):
Speaking [u'blank']
IO - speech.speak (12:40:19):
Speaking [u'Word 2013  subMenu  W  1 of 10']
IO - inputCore.InputManager.executeGesture (12:40:19):
Input: kb(laptop):enter
IO - speech.speak (12:40:20):
Speaking [u'Taskbar']
DEBUGWARNING - RPC process 4000 (WINWORD.EXE) (12:40:25):
Thread 1788, build\x86\remote\ia2Support.cpp, installIA2Support, 86:
Error registering class object, code -2147221008

DEBUGWARNING - watchdog._watcher (12:40:26):
Trying to recover from freeze, core stack:
  File "nvda.pyw", line 191, in <module>
  File "core.pyc", line 381, in main
  File "wx\_core.pyc", line 8657, in MainLoop
  File "wx\_core.pyc", line 7952, in MainLoop
  File "core.pyc", line 352, in Notify
  File "queueHandler.pyc", line 83, in pumpAll
  File "queueHandler.pyc", line 50, in flushQueue
  File "eventHandler.pyc", line 61, in _queueEventCallback
  File "eventHandler.pyc", line 138, in executeEvent
  File "eventHandler.pyc", line 151, in doPreGainFocus
  File "api.pyc", line 107, in setFocusObject
  File "baseObject.pyc", line 34, in __get__
  File "baseObject.pyc", line 110, in _getPropertyViaCache
  File "NVDAObjects\__init__.pyc", line 475, in _get_container
  File "baseObject.pyc", line 34, in __get__
  File "baseObject.pyc", line 110, in _getPropertyViaCache
  File "NVDAObjects\IAccessible\__init__.pyc", line 913, in _get_parent
  File "NVDAObjects\__init__.pyc", line 69, in __call__
  File "NVDAObjects\IAccessible\__init__.pyc", line 608, in __init__
  File "comtypes\__init__.pyc", line 1078, in QueryInterface

DEBUGWARNING - eventHandler.executeEvent (12:40:26):
error executing event: gainFocus on <NVDAObjects.IAccessible.IAccessible object at 0x05E913F0> with extra args of {}
Traceback (most recent call last):
  File "eventHandler.pyc", line 138, in executeEvent
  File "eventHandler.pyc", line 151, in doPreGainFocus
  File "api.pyc", line 107, in setFocusObject
  File "baseObject.pyc", line 34, in __get__
  File "baseObject.pyc", line 110, in _getPropertyViaCache
  File "NVDAObjects\__init__.pyc", line 475, in _get_container
  File "baseObject.pyc", line 34, in __get__
  File "baseObject.pyc", line 110, in _getPropertyViaCache
  File "NVDAObjects\IAccessible\__init__.pyc", line 913, in _get_parent
  File "NVDAObjects\__init__.pyc", line 76, in __call__
  File "NVDAObjects\IAccessible\__init__.pyc", line 385, in findOverlayClasses
  File "baseObject.pyc", line 34, in __get__
  File "baseObject.pyc", line 110, in _getPropertyViaCache
  File "watchdog.pyc", line 198, in _COMError_init
CallCancelled
IO - speech.speak (12:40:27):
Speaking [u'Documento1 - Word']
IO - speech.speak (12:40:27):
Speaking [u'Documento1']
IO - speech.speak (12:40:27):
Speaking [u'Documento de Microsoft Word  edit  multi line']
IO - speech.speak (12:40:27):
Speaking [u'page 1', u'blank']
IO - speech.speak (12:40:30):
Speaking [u'Fichas de la cinta de opciones  tab control']
IO - speech.speak (12:40:30):
Speaking [u'Inicio  tab  Alt, O']
IO - inputCore.InputManager.executeGesture (12:40:31):
Input: kb(laptop):rightArrow
IO - speech.speak (12:40:31):
Speaking [u'Insertar  tab  Alt, B']
IO - inputCore.InputManager.executeGesture (12:40:33):
Input: kb(laptop):enter
IO - inputCore.InputManager.executeGesture (12:40:33):
Input: kb(laptop):downArrow
IO - inputCore.InputManager.executeGesture (12:40:34):
Input: kb(laptop):downArrow
IO - inputCore.InputManager.executeGesture (12:40:35):
Input: kb(laptop):downArrow
IO - inputCore.InputManager.executeGesture (12:40:36):
Input: kb(laptop):rightArrow
IO - inputCore.InputManager.executeGesture (12:40:37):
Input: kb(laptop):leftArrow
IO - inputCore.InputManager.executeGesture (12:40:40):
Input: kb(laptop):escape
DEBUGWARNING - IAccessibleHandler.accessibleObjectFromEvent (12:40:40):
oleacc.AccessibleObjectFromEvent with window 131838, objectID 14 and childID 0: [Error -2147467259] Error no especificado
IO - speech.speak (12:40:40):
Speaking [u'pane']
IO - inputCore.InputManager.executeGesture (12:40:41):
Input: kb(laptop):escape
DEBUGWARNING - IAccessibleHandler.accessibleObjectFromEvent (12:40:41):
oleacc.AccessibleObjectFromEvent with window 131838, objectID 16 and childID 0: [Error -2147467259] Error no especificado
IO - speech.speak (12:40:41):
Speaking [u'Documento1']
IO - speech.speak (12:40:41):
Speaking [u'Documento de Microsoft Word  edit  multi line']
IO - speech.speak (12:40:41):
Speaking [u'page 1', u'blank']
IO - inputCore.InputManager.executeGesture (12:40:44):
Input: kb(laptop):alt+b
IO - inputCore.InputManager.executeGesture (12:40:47):
Input: kb(laptop):escape
DEBUGWARNING - IAccessibleHandler.accessibleObjectFromEvent (12:40:47):
oleacc.AccessibleObjectFromEvent with window 131838, objectID 18 and childID 0: [Error -2147467259] Error no especificado
IO - speech.speak (12:40:47):
Speaking [u'Fichas de la cinta de opciones  tab control']
IO - speech.speak (12:40:47):
Speaking [u'Insertar  tab  Alt, B']
IO - inputCore.InputManager.executeGesture (12:40:49):
Input: kb(laptop):escape
DEBUGWARNING - IAccessibleHandler.accessibleObjectFromEvent (12:40:49):
oleacc.AccessibleObjectFromEvent with window 131962, objectID -4 and childID 0: [Error -2147024809] El parámetro no es correcto
DEBUGWARNING - IAccessibleHandler.accessibleObjectFromEvent (12:40:49):
oleacc.AccessibleObjectFromEvent with window 131838, objectID 19 and childID 0: [Error -2147467259] Error no especificado
IO - speech.speak (12:40:49):
Speaking [u'Documento1']
IO - speech.speak (12:40:49):
Speaking [u'Documento de Microsoft Word  edit  multi line']
IO - speech.speak (12:40:49):
Speaking [u'page 1', u'blank']
IO - inputCore.InputManager.executeGesture (12:40:51):
Input: kb(laptop):NVDA+f1
INFO - globalCommands.GlobalCommands.script_navigatorObject_devInfo (12:40:51):
Developer info for navigator object:
name: u'Documento de Microsoft Word'
role: ROLE_EDITABLETEXT
states: STATE_MULTILINE, STATE_FOCUSABLE, STATE_FOCUSED
isFocusable: True
hasFocus: True
Python object: <NVDAObjects.Dynamic_IAccessibleWordDocumentWindowNVDAObject object at 0x05E32D50>
Python class mro: (<class 'NVDAObjects.Dynamic_IAccessibleWordDocumentWindowNVDAObject'>, <class 'NVDAObjects.IAccessible.IAccessible'>, <class 'NVDAObjects.window.winword.WordDocument'>, <class 'NVDAObjects.behaviors.EditableTextWithoutAutoSelectDetection'>, <class 'editableText.EditableTextWithoutAutoSelectDetection'>, <class 'NVDAObjects.behaviors.EditableText'>, <class 'editableText.EditableText'>, <class 'NVDAObjects.window.Window'>, <class 'NVDAObjects.NVDAObject'>, <class 'baseObject.ScriptableObject'>, <class 'baseObject.AutoPropertyObject'>, <type 'object'>)
description: None
location: (270, 83, 1079, 623)
value: None
appModule: <'appModuleHandler' (appName u'winword', process ID 4000) at address 5e91ff0>
appModule.productName: u'Microsoft Office 2013'
appModule.productVersion: u'15.0.4763.1000'
TextInfo: <class 'NVDAObjects.window.winword.WordDocumentTextInfo'>
windowHandle: 131838
windowClassName: u'_WwG'
windowControlID: 0
windowStyle: 1442840576
windowThreadID: 2188
windowText: u'Documento de Microsoft Word'
displayText: u''
IAccessibleObject: <POINTER(IAccessible) ptr=0x5ea910 at 5e43c60>
IAccessibleChildID: 0
IAccessible event parameters: windowHandle=131838, objectID=-4, childID=0
IAccessible accName: u'Documento de Microsoft Word'
IAccessible accRole: ROLE_SYSTEM_CLIENT
IAccessible accState: STATE_SYSTEM_FOCUSED, STATE_SYSTEM_FOCUSABLE, STATE_SYSTEM_VALID (1048580)
IAccessible accDescription: None
[error file Word 2013.txt](https://github.com/nvaccess/nvda/files/45201/error.file.Word.2013.txt)
IAccessible accValue: None

**Comment by jcsteh** [comment]
Same comments apply as #5546: I can't reproduce this. It's as if this information just never gets to NVDA, which is very odd. Have you tried restarting NVDA with all add-ons disabled?

**Comment by netblue44** [comment]
Yes I have, and I also tried uninstalling all office updates. I remember that this was a problem in one of the 2013 versions of NVDA, and a ticket was filed, and even thow the problem was fixed it was never closed.

**Comment by netblue44** [comment]
Sorry it was a 2014 version the ticket is #4683

**Comment by jcsteh** [comment]
It seems this happens when the ribbon is collapsed/minimised, but not when it is expanded. The workaround for now is to expand the ribbon by pressing control+f1. This does not happen in Word 2010, but does in both 2013 and 2016.

Technical: Brief testing shows this is definitely due to an event being blocked by eventHandler.shouldAcceptEvents. Regression possibly introduced by 46f53f9b or b0eceb5a.

**Comment by netblue44** [comment]
Ok will keep a tab on this.

**Comment by nvaccessAuto** [comment]
Incubated in d849aa542e39d0c009ecfed11da28ee603446e1a.
